Demo/Application/Features/SEPA Direct Debit/BraintreeDemoSEPADirectDebitViewController.swift

```diff
@@ -81,5 +81,65 @@ class BraintreeDemoSEPADirectDebitViewController: BraintreeDemoBaseViewControlle
     private func generateRandomCustomerID() -> String {
         String(UUID().uuidString.replacingOccurrences(of: "-", with: "").prefix(20))
     }
+
+    // MARK: - Helper functions for IBAN generation
+
+    private func generateRandomIBAN() -> String {
+        let countryCode: String = "FR"
+        let bankCode: String = "30006"
+        let branchCode: String = "00001"
+        let accountNumber = Int.random(in: 10_000_000_000...100_000_000_000)
+
+        let accountNumberWithChecksum = accountNumberWithChecksum(bankCode: Int(bankCode) ?? 0, branchCode: Int(branchCode) ?? 0, accountNumber: accountNumber)
+        let checksum = checksum(bankCode: bankCode, branchCode: branchCode, accountNumber: Int(accountNumberWithChecksum) ?? 0)
+
+        return countryCode + "\(checksum)\(bankCode)\(branchCode)\(accountNumberWithChecksum)"
+    }
+
+    private func accountNumberWithChecksum(bankCode: Int, branchCode: Int, accountNumber: Int) -> String {
+        let sum: Int = 89 * bankCode + 15 * branchCode + 3 * accountNumber
+        let checksum = 97 - calculateMod97(from: sum)
+
+        return "\(accountNumber)\(checksum)"
+    }
+
+    private func checksum(bankCode: String, branchCode: String, accountNumber: Int) -> String {
+        // 152700 is taken from the conversion table here: https://community.appway.com/screen/kb/article/generating-and-validating-an-iban-1683400256881#conversion-table
+        // and is representative of the characters "FR" with 00 being added to the end for all bban's to calculate the checksum
+        let bbanString: String = bankCode + branchCode + "\(accountNumber)" + "152700"
+        let modResult = (Decimal(string: bbanString) ?? 0) % 97
```

Comment:
This is where the decimal extension comes into play as we need to divide a 29 character digit which is not possible with Int types. It's also not possible with Decimal by default so there is an extension below that allows us to do calculations on large integers.